src/lib.rs

```diff
@@ -1,3 +1,117 @@
+//! A library to help convert from a unicode string and attributes to a list of glyphs and
+//! positions.
+//!
+//! Converting unicode strings to glyphs is a complex process, requiring knowledge of
+//!  - which fonts to use (and choosing an appropriate font given a size, script, typeface, weight,
+//!    etc.),
+//!  - a description of how to render a font (e.g. a truetype font file)
+//!  - the space available for drawing and what to do if it's not enough,
+//!  - and information on the user's locale.
+//!
+//! # Font terms
+//!
+//! The terminology of printing can be confusing. Partly this is because printing has been so
+//! important for so long, and so has historical baggage, but it is also because unifying the
+//! written methods of many different cultures requires a complicated system with many parts.
+//! Unfortunately, a consequence of this is that most of the terms used in typesetting have
+//! different meanings to different people. The unicode standard is useful in that it precisely
+//! defines the terms it uses, but these meanings aren't necessarily the same as those that are in
+//! common usage.  Below are some important terms in font rendering and the meanings some ascribe
+//! to them:
+//!
+//! This is a work in progress and will be wrong in places (for now).
+//!
+//!  - *typography* - The study of drawing text to paper, screen, or some other medium. Drawing
+//!    text is also known as *typesetting*, or *rendering*.
+//!  - *symbol* - An image drawn or printed on a canvas, sheet of paper, or screen. Physically the
+//!    same as a picture.
+//!  - *glyph* - A symbol from a collection of symbols with some semantic meaning. The collection
+//!    could be a script (like latin - `ABCDE....abcde...`, or it could be from the set of
+//!    mathemtatical symbols (`+-×÷`). In the case of languages, another word more commonly used to
```

Comment:
mathematical

Reply:
Also, because you’ve used the proper symbols × and ÷ I feel like quibbling over the use of - (U+002D HYPHEN-MINUS) where − (U+2212 MINUS SIGN) is available. I’m like that. 😁